pkg/tasks/task_selector.go

```diff
@@ -0,0 +1,54 @@
+package tasks
+
+import (
+	"fmt"
+
+	osidentity "github.com/devbuddy/devbuddy/pkg/helpers/os"
+)
+
+type TaskSelector interface {
+	ShouldRun(*Context, *Task) error
+}
+
+type TaskSelectorImpl struct {
+	osIdent osidentity.Identity
+}
+
+func NewTaskSelector() (*TaskSelectorImpl, error) {
+	osIdent, err := osidentity.Detect()
```

Comment:
I'm wondering if this should be done inside the abstract `NewTaskSelector`? What about doing this inside the implementation, then you would simplify the return value and the `TaskSelectorImpl` would not take any parameters and we would be able to get ride of `NewTaskSelector`?